ocean_lib/models/v4/test/test_erc20_enterprise.py

```diff
@@ -0,0 +1,294 @@
+    balance_opf_publish_before = mock_usdc_contract.balanceOf(opf_collector_address)
+
+    tx = erc20_enterprise_token.buy_from_dispenser_and_order(
+        order_params=order_params,
+        dispenser_address=dispenser.address,
+        from_wallet=consumer_wallet,
+    )
+
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    assert tx_receipt["status"] == 1
+
+    balance_consume = mock_dai_contract.balanceOf(consume_fee_address)
+    balance_opf_consume = mock_dai_contract.balanceOf(opf_collector_address)
+
+    balance_publish = mock_usdc_contract.balanceOf(publish_fees[0])
+    balance_opf_publish = mock_usdc_contract.balanceOf(opf_collector_address)
+
+    expected_consume = consume_fee_amount - consume_fee_amount / 100
+    expected_opf_consume = consume_fee_amount / 100
+
+    expected_publish = publish_fees[2] - publish_fees[2] / 100
+    expected_opf_publish = publish_fees[2] / 100
+
+    assert balance_consume - balance_consume_before == expected_consume
+    assert balance_opf_consume - balance_opf_consume_before == expected_opf_consume
+
+    assert balance_publish - balance_publish_before == expected_publish
+    assert balance_opf_publish - balance_opf_publish_before == expected_opf_publish
+
+    assert (
+        erc20_enterprise_token.balanceOf(erc20_enterprise_token.get_fee_collector())
+        == 0
+    )
+
+
+def test_buy_from_fre_and_order(
+    web3, config, consumer_wallet, publisher_wallet, factory_deployer_wallet
+):
+    mock_usdc_contract = ERC20Token(web3, get_address_of_type(config, "MockUSDC"))
+    mock_dai_contract = ERC20Token(web3, get_address_of_type(config, "MockDAI"))
+    fixed_rate_exchange = FixedRateExchangeV4(
+        web3, get_address_of_type(config, "FixedPrice")
+    )
+
+    _, erc20_enterprise_token = deploy_erc721_erc20(
+        web3=web3,
+        config=config,
+        cap=web3.toWei(100, "ether"),
+        erc20_minter=consumer_wallet,
+        erc721_publisher=consumer_wallet,
+        template_index=2,
+    )
+    erc20_enterprise_token = ERC20Enterprise(web3, erc20_enterprise_token.address)
+
+    fixed_data = FixedData(
+        fixed_price_address=fixed_rate_exchange.address,
+        addresses=[
+            mock_usdc_contract.address,
+            consumer_wallet.address,
+            consumer_wallet.address,
+            ZERO_ADDRESS,
+        ],
+        uints=[18, 18, web3.toWei(1, "ether"), web3.toWei(1, "ether"), 1],
+    )
+
+    tx = erc20_enterprise_token.create_fixed_rate(
+        fixed_data=fixed_data,
+        from_wallet=consumer_wallet,
+    )
+
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    assert tx_receipt["status"] == 1
+
+    new_fixed_rate_event = erc20_enterprise_token.get_event_log(
+        "NewFixedRate",
+        from_block=tx_receipt.blockNumber,
+        to_block=web3.eth.block_number,
+        filters=None,
+    )
+
+    exchange_id = new_fixed_rate_event[0]["args"]["exchangeId"]
```

Comment:
you can access the `exchange_id` with `new_fixed_rate_event[0].args.exchangeId`.